### PR TITLE
node: cleanup construction

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -345,50 +337,29 @@ func makeNode(cfg *config.Config,
...
-	// We need to set Seeds and PersistentPeers on the switch,
-	// since it needs to be able to use these (and their DNS names)
-	// even if the PEX is off. We can include the DNS name in the NetAddress,
-	// but it would still be nice to have a clear list of the current "PersistentPeers"
-	// somewhere that we can return with net_info.
-	//
-	// If PEX is on, it should handle dialing the seeds. Otherwise the switch does it.
-	// Note we currently use the addrBook regardless at least for AddOurAddress
-
 	pexReactor, err := createPEXReactor(logger, peerManager, router)
```
Comment (cmwaters) [inline node/node.go:358]: The pex reactor is something we can still turn off right? Do we check the config anywhere? Or is it just `OnStart()`

File: node/node.go [hunk 2]
```diff
@@ -522,6 +482,14 @@ func makeSeedNode(cfg *config.Config,
 
 // OnStart starts the Node. It implements service.Service.
 func (n *nodeImpl) OnStart() error {
+
+	if n.config.RPC.PprofListenAddress != "" {
+		go func() {
+			logger.Info("Starting pprof server", "laddr", n.config.RPC.PprofListenAddress)
+			logger.Error("pprof server error", "err", http.ListenAndServe(n.config.RPC.PprofListenAddress, nil))
+		}()
+	}
```
Comment (cmwaters) [inline node/node.go:491]: I like this change 👍 Are we able to turn it off `OnStop()`?

**Review by cmwaters — APPROVED** [review]

**Review by creachadair — APPROVED** [review]

File: node/node.go (hunk 2 above)
Comment (tychoish) [reply node/node.go:491]: I don't believe that's a thing that's currently supported, we certainly never have before.

File: node/node.go (hunk 1 above)
Comment (tychoish) [reply node/node.go:358]: That configuration value isn't checked anywhere in the current iteration of the code.

Comment (cmwaters) [reply node/node.go:358]: You mean `Config.P2P.PexReactor` isn't used at all? Can we change that and backport it to v0.35? Pex should be configurable

Comment (tychoish) [reply node/node.go:358]: #7198

**Pull request merged**